myria/relation.py

```diff
@@ -0,0 +1,71 @@
+from dateutil.parser import parse
+from itertools import izip
+from myria import MyriaConnection, MyriaError
+from schema import MyriaSchema
+
+class MyriaRelation(object):
+  DefaultConnection = MyriaConnection(hostname='localhost', port=8753)
+
+  # MyriaRelation({'userName': 'public', 'programName': 'adhoc', 'relationName': 'relation'})
+  # or
+  # MyriaRelation('public:adhoc:relation')
+  # or
+  # MyriaRelation('relation') # Defaults are 'public' and 'adhoc'
+  def __init__(self, relation, connection=DefaultConnection, *args, **kwargs):
+    self.name = relation if isinstance(relation, basestring) else relation.name
+    self.components = self._get_name_components(self.name)
+    self.connection = connection
+    self.qualified_name = self._get_qualified_name(self.components)
+    self._schema = kwargs['schema'] if 'schema' in kwargs else None
+
+    # Should probably expose a better way of creating "new" relations
+    if not self._schema is None and self._has_metadata:
+      raise MyriaError('New relation specified (schema != None), but it already exists on the server.')
+
+  def toJson(self):
+    return self.connection.download_dataset(self.qualified_name)
+
+  @property 
+  def schema(self):
+    if self._schema is None:
+      self._schema = MyriaSchema(json=self._metadata['schema'])
+    return self._schema
+
+  @property
+  def createdDate(self):
+    return parse(self._metadata['created'])
+
+  def __len__(self):
+    return int(self._metadata['numTuples'])
+
+  @property
+  def _metadata(self):
+    if 'metadata' not in self.__dict__:
```

Comment:
I'd be okay with that as well. Setting metadata to None initially is more explicit, though.